src/Microsoft.ML.Tokenizers/Model/BPETokenizer.cs

```diff
@@ -1063,26 +1420,68 @@ private int EncodeToIdsWithCache(ReadOnlySpan<char> text, List<int>? accumulated
 
             Word word;
 
+            scoped ReadOnlySpan<char> textSpan = text;
+
+            Span<char> token = stackalloc char[BufferLength];
+            Span<int> mapping = stackalloc int[BufferLength];
```

Comment:
Instead of doing these here, would it be better to add an `else` to the if statement where you use the array pool and only do this stackalloc if you need it?

I won't comment this on all the places that do this, but if this would change here we should change it everywhere.